### ISSUE TITLE
Add Twitter Referrer

### ISSUE THREAD

**Issue by swapagarwal** [swapagarwal/mediumship#5]
https://twitter.com/ev/status/1100899021621583872

**Comment by shridharkapshikar** [comment]
Swpanil, please correct me.  When I am trying to login to browser without adding an extension, it shows me 2 options, 1 - login with Google account and 2 -  login with Facebook account. So Twitter referrer should be added on that window? please confirm. 
Also, when I add the Medium extension to chrome, and I can see "Sign in with Twitter"

**Comment by rtvkiz** [comment]
Yes Twitter referrer should be added to the login page

**Comment by Stevo99** [comment]
I can work on this if it is still open.  Let me know thanks!

**Comment by j3soon** [comment]
@Stevo99 The method we're using now is find its canonical url and open it in a new window. If the canonical url doesn't exist, we'll open it in incognito mode. The 2 methods combined works in most of the cases. 

```js
/* chrome */
chrome.windows.create({"url": webCanonicalUrl});
chrome.windows.create({"url": url, "incognito": true});
/* firefox */
browser.tabs.create({"url": webCanonicalUrl});
browser.windows.create({"url": url, "incognito": true});
```

We would appreciate if you could somehow add the twitter referrer. Currently, @swapagarwal maintains the chrome version, I maintain the firefox version, and we both haven't work on this feature yet.

**Comment by swapagarwal** [comment]
Resolved in dd960d280997f67d98c5323f11e32979f9f1062a
